tests/test_get_parser_unit.py

```diff
@@ -0,0 +1,188 @@
+import os
+import sys
+import time
+import pytest
+
+#
+# we assume everwhere our current directory is in the package 
+# test area, so go ahead and cd there
+#
+os.chdir(os.path.dirname(__file__))
+
+
+#
+# import modules we need to test, since we chdir()ed, can use relative path
+#
+sys.path.append("../lib")
+import get_parser
+
+from test_unit import TestUnit 
+
+@pytest.fixture
+def find_all_arguments():
+    # try to extract all the --foo arguments from the source
+    # and track which ones are flags
+    f = open("../lib/get_parser.py","r")
+    flagargs = set()
+    listargs = set()
+    allargs = []
+    dest = {}
+    for line in f.readlines():
+        p = line.find('"--')
+        mq = '"'
+        if p < 0: 
+           p = line.find("'--")
+           mq = "'"
+
+        if p > 0:
```

Comment:
Whoo - this is a doozy, but short of using regexes, I can't see anything wrong with testing for both the `"` and `'` cases explicitly.

Would you mind adding a quick one-line comment for each `if` statement in here just so we know, 2 years from now, what these clauses are doing (e.g. lines 41-42 are great)?  e.g. `If we find "dest=", save the value in dest dict`.  -d and -f you could probably explain together, as well as the store and append clauses.  I'm just worried about us forgetting what all this does a while from now.

Reply:
Okay, I've added more blow-by-blow comments to the get_parser tests. It was a good exercise, I figured
out I could drop a 4-line clause in the code :-).